run-checks

```diff
@@ -156,30 +230,91 @@ missing_interface_spread_test() {
     done
 }
 
+CURRENTTRAP="true"
+EXIT_CODE=99
+
+store_exit_code() {
+    EXIT_CODE=$?
+}
+
+exit_with_exit_code() {
+    exit $EXIT_CODE
+}
+
+addtrap() {
```

Comment:
I'm inclined to say let's kill this script and replace it with python/make that is not re-inventing the wheel and does not rely on hacks like that.